NetSDK/Services/Cache/Classes/SegmentCache.cs

```diff
@@ -0,0 +1,64 @@
+﻿using Splitio.Services.Cache.Interfaces;
+using Splitio.Services.SegmentFetcher.Classes;
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace Splitio.Services.Cache.Classes
+{
+    public class SegmentCache : ISegmentCache
+    {
+        private ConcurrentDictionary<string, SelfRefreshingSegment> segments;
+
+        public void AddToSegment(string segmentName, HashSet<string> segmentKeys)
+        {
+            SelfRefreshingSegment segment;
+            if (segments.TryGetValue(segmentName, out segment))
+            {
+                segment.AddKeys(segmentKeys);
+            }
+        }
+
+        public void RemoveFromSegment(string segmentName, HashSet<string> segmentKeys)
+        {
+            SelfRefreshingSegment segment;
+            if (segments.TryGetValue(segmentName, out segment))
+            {
+                segment.RemoveKeys(segmentKeys);
+            }
+        }
+
+        public bool IsInSegment(string segmentName, string key)
+        {
+            SelfRefreshingSegment segment;
+            if (segments.TryGetValue(segmentName, out segment))
+            {
+                return segment.Contains(key);
+            }
+
+            return false;
+        }
+
+        public void SetChangeNumber(string segmentName, long changeNumber)
+        {
+            SelfRefreshingSegment segment;
+            if (segments.TryGetValue(segmentName, out segment))
+            {
+                segment.changeNumber = changeNumber;
+            }
+        }
+
+        public long GetChangeNumber(string segmentName)
+        {
+            SelfRefreshingSegment segment;
+            if (segments.TryGetValue(segmentName, out segment))
+            {
+                return segment.changeNumber;
```

Comment:
Did you mean, here?
https://github.com/splitio/.net-client/pull/21/files/fbb06b2f094f6051fb253dd952b8ab0cc78c471d#diff-85d0a171f6e3f1d2ea253b8175ba6864R20